frontend/app/src/modules/history/management/forms/swap/SwapSubEventList.vue

```diff
@@ -39,23 +41,58 @@ function remove(index: number) {
 function add() {
   set(modelValue, [...get(modelValue), { amount: '', asset: '' }]);
 }
+
+interface SwapSubEventRef extends ComponentPublicInstance {
+  submitPrice: () => Promise<any>;
+}
+
+const subEventRefs = ref<SwapSubEventRef[]>([]);
+
+function isSwapSubEventComponent(el: Element | ComponentPublicInstance | null): el is SwapSubEventRef {
+  return el !== null
+    && typeof el === 'object'
+    && '$el' in el
+    && 'submitPrice' in el
+    && typeof (el as any).submitPrice === 'function';
+}
```

Comment:
There is not better way for this right? :cry:

Reply:
sadly I don't see better way, either this, or using `as`